### DISCUSSION TITLE
structured dict macro

### DISCUSSION THREAD

**Discussion by off-by-one** [hylang/hy#2162] [Show and tell]
I found the default dict syntax a bit hard to read, but the macros made it really easy to introduce custom syntax. You can do it with a one-liner (fixed thanks to Kodiologist):

```hy
(<defmacro/deftag> <name> [d] `{~@(do (setv l []) (for [(, k _ v) d] (+= l [k v])) l)})
```

And can apply the macro to `{(1 => 2) (3 => 4)}` (or if you prefer Python style, `{(1: 2) (3: 4)}`).

I like `deftag %` (or `defmacro "#%"` in 1.0), since it gives you `#%{(1 => 2) (3 => 4)}`, which is similar to Perl and also suggestively close to the set notation.

**Comment by Kodiologist** [comment]
I think your code will do the wrong thing if there are collection-type models in the arguments, as in `{(1 => 2) (3 => [4 5 6])}`, because `flatten` is recursive.

For what it's worth, I like to put two spaces between key-value pairs in dictionaries, so it's clearer what belongs to what:

```
(dict  :a 1  :b 2  :c 3)
{1 10  2 20  3 30}
```

**Reply by gilch** [reply]
#1069
